code/game/turfs/simulated/floor/lava_pool.dm

```diff
@@ -0,0 +1,59 @@
+/turf/simulated/floor/lava_pool
+		icon_state = "lava"
+		name = "lava pool"
+		desc = "A deadly pool of molten rock that incinerates almost anything on contact."
+		luminosity = 3
+		var/list/blacklist = list(/obj/item/clothing/suit/space/hardsuit/mining, /obj/item/asteroid/goliath_hide)
+		var/processing = 0
+
+/turf/simulated/floor/lava_pool/Entered(var/mob/living/carbon/C)
+	Incinerate(C) //To catch items before other things get processed
+	if(!istype(C))
+		return
+	if(!processing)
+		SSobj.processing |= src
+		processing = 1
+	if(istype(C, /mob/living/carbon/human))
+		if(check_suit(C))
+			return
+
+	apply_lava_damage(C)
+	C.emote("scream")
+	C.visible_message("<span class='warning'>[C] falls into [src]!   </span>", "<span class='userdanger'>You fall into [src]!   </span>", "<span class='warning'>You hear a splash followed by a howling scream! </span>")
+
+/turf/simulated/floor/lava_pool/proc/process()
+	var/mobpresent = 0
+	for(var/mob/living/carbon/human/H in src)
+		mobpresent = 1
+		if(check_suit(H))
+			continue
```

Comment:
@Stickymayhem @Miauw62 is right, Carbons will be burned on Entered() but not in process().
your process loop should be:
```
var/mobpresent = 0
for(var/mob/living/carbon/C in src)
    mobpresent = 1
    if(istype(C, /mob/living/carbon/human))
        var/mob/living/carbon/human/H = C
        if(check_suit(H))
           continue
    apply_lava_damage(H)
if(!mobpresent)
    SSobj.processing.Remove(src)
    processing = 0
````